compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -54,6 +54,11 @@ impl<'a> Parser<'a> {
             // that starts like a path (1 token), but it fact not a path.
             // Also, we avoid stealing syntax from `parse_item_`.
             self.parse_stmt_path_start(lo, attrs)?
+        } else if self.check_inline_const() {
+            let const_span = lo.to(self.token.span);
+            let kind = ExprKind::AnonConst(self.parse_anon_const(const_span)?);
+            let e = self.mk_expr(const_span, kind, attrs.into());
+            self.mk_stmt(lo.to(e.span), StmtKind::Expr(e))
```

Comment:
`1 + const { 2 }` isn't going to work like this.
Parsing of const blocks themselves need to go into `compiler\rustc_parse\src\parser\expr.rs`, roughly to the same place where `unsafe { ... }` blocks are parsed.

`parse_stmt_without_recovery` (or maybe `parse_item_kind`) should only ensure that we don't start parsing const block as an item.